bookkeeper-server/src/main/java/org/apache/bookkeeper/client/api/LedgerMetadata.java

```diff
@@ -0,0 +1,122 @@
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.bookkeeper.client.api;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Map;
+import java.util.NavigableMap;
+import org.apache.bookkeeper.common.annotation.InterfaceAudience.LimitedPrivate;
+import org.apache.bookkeeper.common.annotation.InterfaceStability.Unstable;
+import org.apache.bookkeeper.net.BookieSocketAddress;
+
+/**
+ * Represents the client-side metadata of a ledger. It is immutable.
+ *
+ * @since 4.6
+ */
+@LimitedPrivate
+@Unstable
+public interface LedgerMetadata extends Serializable {
+
+    /**
+     * Returns the ensemble size of this ledger.
+     *
+     * @return the ensemble size of this ledger.
+     */
+    int getEnsembleSize();
+
+    /**
+     * Returns the write quorum size of this ledger.
+     *
+     * @return the write quorum size of this ledger.
+     */
+    int getWriteQuorumSize();
+
+    /**
+     * Returns the ack quorum size of this ledger.
+     *
+     * @return the ack quorum size of this ledger.
+     */
+    int getAckQuorumSize();
+
+    /**
+     * Returns the last entry id of this ledger.
+     *
+     * <p>If this ledger is not sealed {@link #isSealed()}, it returns {@code -1L}.
+     *
+     * @return the last entry id of this ledger if it is sealed, otherwise -1.
+     */
+    long getLastEntryId();
+
+    /**
+     * Returns the length of this ledger.
+     *
+     * <p>If this ledger is not sealed {@link #isSealed()}, it returns {@code 0}.
+     *
+     * @return the length of this ledger if it is sealed, otherwise 0.
+     */
+    long getLength();
+
+    /**
+     * Returns the digest type used by this ledger.
+     *
+     * @return the digest type used by this ledger.
+     */
+    DigestType getDigestType();
+
+    /**
+     * Returns the creation timestamp of this ledger.
+     *
+     * @return the creation timestamp of this ledger.
+     */
+    long getCtime();
+
+    /**
+     * Returns whether the ledger is sealed or not.
+     *
+     * @return true if the ledger is sealed, otherwise false.
+     */
+    boolean isSealed();
+
+    /**
+     * Returns the custom metadata stored with the ledgers.
+     *
+     * @return the custom metadata stored with the ledgers.
+     */
+    Map<String, byte[]> getCustomMetadata();
+
+    /**
+     * Returns the ensemble at the given {@code entryId}.
+     *
+     * @param entryId the entry id to retrieve its ensemble information
+     * @return the ensemble which contains the given {@code entryId}.
+     */
+    List<BookieSocketAddress> getEnsembleAt(long entryId);
```

Comment:
is BookieSocketAddress in the new API ?

Reply:
`BookieSocketAddress` is under `net` package not `client` package. it is a common class.

I think we need to move `net` package to bookkeeper-common at some point. If you prefer me doing this movement in this pr, I can do that.

Reply:
Let's keep it here now. The module is not important, until we modularize bk in distinct  java 9 modules.
It is important that now that package is in the new public API and we won't change/hide it in the future.
An alternative would be introduce an interface and we are doing for Ledger Metadata. In the future the full .client.api package will have its own module.

Reply:
@eolivelli BookieSocketAddress is shared by both client and server. I feel it is a common class, instead of an interface part of `client.api` package.